### PR TITLE
python3Packages.flask-paranoid: init at 0.2

### PR DESCRIPTION
**Pull request by gador**
Init of Flask-Paranoid

###### Motivation for this change
needed for pgadmin4 init
https://github.com/NixOS/nixpkgs/pull/154764

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) 
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4262,6 +4262,16 @@
...
+  gador = {
+    email = "florian.brandes@posteo.de";
+    github = "gador";
+    githubId = 1883533;
+    name = "Florian Brandes";
+    keys = [{
+      longkeyid = "rsa4096/0xBBB3E40E53797FD9";
+      fingerprint = "0200 3EF8 8D2B CF2D 8F00  FFDC BBB3 E40E 5379 7FD9";
+    }];
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4274]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/python-modules/flask-paranoid/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  propagatedBuildInputs = [
+    flask
+  ];
+
+  pythonImportsCheck = [ "flask_paranoid" ];
+
+  # PyPi version does not include test directory
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-paranoid/default.nix:24]: Can we fetch from GitHub and execute tests? They tag releases https://github.com/miguelgrinberg/flask-paranoid/tree/v0.2.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The init commit should be ``python3Packages.flask-paranoid: init at 0.2.0`` or 0.2 based on the git tag.

**Comment by gador** [comment]
@ofborg eval

**Comment by gador** [comment]
@ofborg test

**Comment by gador** [comment]
@ofborg test tarball.nixpkgs-basic-release-checks

File: pkgs/development/python-modules/flask-paranoid/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "flask-paranoid";
+  version = "v0.2";
+
+  src = fetchFromGitHub {
+    owner = "miguelgrinberg";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-paranoid/default.nix:15]:
```suggestion
  version = "0.2";

  src = fetchFromGitHub {
    owner = "miguelgrinberg";
    repo = pname;
    rev = "v${version}";
```

File: pkgs/development/python-modules/flask-paranoid/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "flask_paranoid" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/miguelgrinberg/flask-paranoid/";
+    description = "Simple user session protection";
+    license = licenses.mit;
+    maintainers = with maintainers; [ gador ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-paranoid/default.nix:39]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Or you might need to rebase this branch on master if you already added yourself.

File: pkgs/development/python-modules/flask-paranoid/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "miguelgrinberg";
+    repo = pname;
+    rev = version;
+    sha256 = "0WWc/ktAOuTk4A75xI1jCj/aef2+1TjLKBA9+PRfJO0=";
+  };
+
+  postPatch = ''
+    # tests have a typo in one of the assertions
+    # fixing this here
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-paranoid/default.nix:21]:
```suggestion
```

File: pkgs/development/python-modules/flask-paranoid/default.nix (hunk 4 above)
Comment (gador) [reply pkgs/development/python-modules/flask-paranoid/default.nix:39]: Its already added. I rebased this branch on the current master

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 155215` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python310Packages.flask-paranoid</li>
<li>python39Packages.flask-paranoid</li>
</ul>
</details>
